ui/lib/kv/addon/components/page/secret/details.hbs

```diff
@@ -2,65 +2,77 @@
   <:tabLinks>
     <LinkTo @route="secret.details" data-test-secrets-tab="Secret">Secret</LinkTo>
     <LinkTo @route="secret.metadata.index" data-test-secrets-tab="Metadata">Metadata</LinkTo>
-    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Version History">Version History</LinkTo>
-    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Version Diff">Version Diff</LinkTo>
+    {{#unless @noMetadataPermission}}
+      <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Version History">Version History</LinkTo>
+      <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Version Diff">Version Diff</LinkTo>
+    {{/unless}}
   </:tabLinks>
 
   <:toolbarFilters>
-    <Toggle
-      data-test-json-view-toggle
-      @name="json"
-      @status="success"
-      @size="small"
-      @checked={{this.showJsonView}}
-      @onChange={{this.toggleJsonView}}
-    >
-      <span class="has-text-grey">JSON</span>
-    </Toggle>
+    {{#unless this.emptyState}}
+      <Toggle
+        data-test-json-view-toggle
+        @name="json"
+        @status="success"
+        @size="small"
+        @checked={{this.showJsonView}}
+        @onChange={{this.toggleJsonView}}
+      >
+        <span class="has-text-grey">JSON</span>
+      </Toggle>
+    {{/unless}}
   </:toolbarFilters>
   <:toolbarActions>
     <ToolbarLink @route="secret.edit" @type="add">Create new version</ToolbarLink>
   </:toolbarActions>
 </KvPageHeader>
 
-<div class="info-table-row-header">
-  <div class="info-table-row thead {{if this.showJsonView 'is-shadowless'}} ">
-    {{#unless this.showJsonView}}
-      <div class="th column is-one-quarter">
-        Key
-      </div>
-      <div class="th column">
-        Value
+{{#if this.emptyState}}
+  <EmptyState @title={{this.emptyState.title}} @message={{this.emptyState.message}}>
+    {{#if this.emptyState.link}}
+      <DocLink @path={{this.emptyState.link}}>Learn more</DocLink>
+    {{/if}}
+  </EmptyState>
+{{else}}
+  <div class="info-table-row-header">
+    <div class="info-table-row thead {{if this.showJsonView 'is-shadowless'}} ">
+      {{#unless this.showJsonView}}
+        <div class="th column is-one-quarter">
+          Key
+        </div>
+        <div class="th column">
+          Value
+        </div>
+      {{/unless}}
+      <div class="th column justify-right" data-test-created-time>
+        {{#if @secret.createdTime}}
```

Comment:
Looking at the API it looks like as long as we've returned the metadata object on the the data/ endpoint we should see createdTime and version. Do we need this conditional, or are there times we may not return the metadata object? And if there are times, then maybe make the conditional clearer like `#if @secret.metadata`

Reply:
Hmm... found [your old PR ](https://github.com/hashicorp/vault/pull/19971/files)that had a bug fix for just this situation. Do you remember how you could get to a spot where the API didn't return a createdTime?

Reply:
That was because `SecretFormShow` rendered for not just KV secrets (it looks like I added a test for cubbyhole.

Since this will just be a component in KV, the conditional can be removed if `created_time` is always present!